File: OdontoFlowWeb/OdontFlow/src/app/core/model/employee-view.model.ts
export interface EmployeeViewModel {
    id?: string;
    name?: string;
    email?: string;
    applyCommission?: boolean;
    isSalesRepresentative?: boolean;
    commissionPercentage?: number;
    active?: boolean; 
  }
  
  export interface EmployeeClientSummaryViewModel {
    id: string;
    name: string;
    commissionPercentage: number; // Comisión personalizada
  }

  export interface CreateEmployeeDto {
    name?: string;
    email?: string;
    applyCommission: boolean;
    isSalesRepresentative: boolean;
    commissionPercentage: number;
    active?: boolean; 
  }

  export interface UpdateEmployeeDto extends CreateEmployeeDto {
    id: string | undefined;
  } 

  export interface CreateClientCommissionDto {
    clientId: string;
    commissionPercentage: number;
  }
  

File: OdontoFlowAPI/OdontFlow/OdontFlow.Domain/ViewModel/Employee/EmployeeViewModel.cs
namespace OdontFlow.Domain.ViewModel.Employee;

public class EmployeeViewModel
{
    public Guid Id { get; set; }
    public string Name { get; set; } = default!;
    public string Email { get; set; } = default!;
    public bool ApplyCommission { get; set; }
    public bool IsSalesRepresentative { get; set; }
    public float CommissionPercentage { get; set; }
    public bool Active { get; set; }
    public List<EmployeeClientSummaryViewModel>? Clients { get; set; }
}

 